packages/alfa-rules/src/common/predicate/is-offscreen.ts

```diff
@@ -0,0 +1,46 @@
+import { Cache } from "@siteimprove/alfa-cache";
+import { Device } from "@siteimprove/alfa-device";
+import { Element, Node } from "@siteimprove/alfa-dom";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Context } from "@siteimprove/alfa-selector";
+import { Style } from "@siteimprove/alfa-style";
+
+const { abs } = Math;
+const { isElement } = Element;
+
+const cache = Cache.empty<Device, Cache<Context, Cache<Node, boolean>>>();
+
+export function isOffscreen(
+  device: Device,
+  context: Context = Context.empty()
+): Predicate<Node> {
+  return function isOffscreen(node) {
+    return cache
+      .get(device, Cache.empty)
+      .get(context, Cache.empty)
+      .get(node, () => {
+        if (isElement(node)) {
+          const style = Style.from(node, device, context);
+
+          const { value: position } = style.computed("position");
+          const { value: left } = style.computed("left");
+          const { value: right } = style.computed("right");
+
+          if (position.value !== "static") {
+            for (const inset of [left, right]) {
+              switch (inset.type) {
+                case "length":
+                  return abs(inset.value) >= 9999;
+              }
```

Comment:
Why not looking also at `top` (I guess `bottom` is less doable due to infinity vertical scrolling)? (`position: absolute; top: -9999px` seems to be the preferred offscreen way in ACT rules at least).

Reply:
As most pages grow downwards, not sideways, content should be offscreen'ed using `left` or `right` as the width of the viewport can be expected to remain constant. This article also talks about the perils of offscreen'ing using `top` and uses `left` in its examples: http://accessibilitytips.com/2008/03/04/positioning-content-offscreen/. This article from WebAIM uses `left` as well: https://webaim.org/techniques/css/invisiblecontent/